internal/status/backendtlspolicyconditions.go

```diff
@@ -0,0 +1,162 @@
+}
+
+// AddCondition adds a condition with the given properties to the
+// BackendTLSPolicyAncestorStatus.
+func (b *BackendTLSPolicyAncestorStatusUpdate) AddCondition(conditionType gatewayapi_v1alpha2.PolicyConditionType, status metav1.ConditionStatus, reason gatewayapi_v1alpha2.PolicyConditionReason, message string) metav1.Condition {
+	var pas *gatewayapi_v1alpha2.PolicyAncestorStatus
+
+	for _, v := range b.PolicyAncestorStatuses {
+		if v.AncestorRef == b.ancestorRef {
+			pas = v
+			break
+		}
+	}
+
+	if pas == nil {
+		pas = &gatewayapi_v1alpha2.PolicyAncestorStatus{
+			AncestorRef:    b.ancestorRef,
+			ControllerName: b.GatewayController,
+		}
+
+		b.PolicyAncestorStatuses = append(b.PolicyAncestorStatuses, pas)
+	}
+
+	idx := -1
+	for i, c := range pas.Conditions {
+		if c.Type == string(conditionType) {
+			idx = i
+			break
+		}
+	}
+
+	if idx > -1 {
+		message = pas.Conditions[idx].Message + ", " + message
+	}
+
+	cond := metav1.Condition{
+		Reason:             string(reason),
+		Status:             status,
+		Type:               string(conditionType),
+		Message:            message,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		ObservedGeneration: b.Generation,
+	}
+
+	if idx > -1 {
+		pas.Conditions[idx] = cond
+	} else {
+		pas.Conditions = append(pas.Conditions, cond)
+	}
+
+	return cond
+}
+
+// ConditionExists returns whether or not a condition with the given type
+// exists.
+func (b *BackendTLSPolicyAncestorStatusUpdate) ConditionExists(conditionType gatewayapi_v1alpha2.PolicyConditionType) bool {
+	for _, c := range b.ConditionsForAncestorRef(b.ancestorRef) {
+		if c.Type == string(conditionType) {
+			return true
+		}
+	}
+	return false
+}
+
+// ConditionsForAncestorRef returns the list of conditions for a given ancestor
+// if it exists.
+func (b *BackendTLSPolicyStatusUpdate) ConditionsForAncestorRef(ancestorRef gatewayapi_v1beta1.ParentReference) []metav1.Condition {
+	for _, pas := range b.PolicyAncestorStatuses {
+		if pas.AncestorRef == ancestorRef {
+			return pas.Conditions
+		}
+	}
+
+	return nil
+}
+
+func (b *BackendTLSPolicyStatusUpdate) Mutate(obj client.Object) client.Object {
+	o, ok := obj.(*gatewayapi_v1alpha2.BackendTLSPolicy)
+	if !ok {
+		panic(fmt.Sprintf("Unsupported %T object %s/%s in status mutator",
+			obj, b.FullName.Namespace, b.FullName.Name,
+		))
+	}
+
+	var newPolicyAncestorStatuses []gatewayapi_v1alpha2.PolicyAncestorStatus
+	for _, pas := range b.PolicyAncestorStatuses {
+		for i := range pas.Conditions {
+			cond := &pas.Conditions[i]
+
+			cond.ObservedGeneration = b.Generation
+			cond.LastTransitionTime = b.TransitionTime
+		}
+
+		newPolicyAncestorStatuses = append(newPolicyAncestorStatuses, *pas)
+	}
+
+	btp := o.DeepCopy()
+
+	// Get all the PolicyAncestorStatuses that are for other Gateways.
+	for _, pas := range o.Status.Ancestors {
+		if !gatewayapi.IsRefToGateway(pas.AncestorRef, b.GatewayRef) {
+			newPolicyAncestorStatuses = append(newPolicyAncestorStatuses, pas)
```

Comment:
just going by the coverage report, I guess we could add another test to make sure existing ancestor statuses from other controllers are preserved

not sure if this is actually properly tested well for the other instances of the *StatusUpdate pattern but we can follow up on those later